src/rest_vol.c

```diff
@@ -1637,6 +1637,53 @@ RV_parse_response(char *HTTP_response, void *callback_data_in, void *callback_da
     return ret_value;
 } /* end RV_parse_response() */
 
+/*-------------------------------------------------------------------------
+ * Function:    RV_copy_object_URI_and_domain_callback
+ *
+ * Purpose:     Copies information from response to provided buffers.
+ * 
+ *              This function should be used in place of 
+ *              RV_copy_object_URI_callback wherever the object
+ *              may be an external link, whose domain is different
+ *              from the domain of its parent object.
+ * 
+ *              callback_data_out is expected to be a pointer to an array
+ *              of two pointers, the first being the address of a buffer for the URI,
+ *              and the second being the address of a pointer to a domain buffer of the target object.
+ * 
+ *              callback_data_is provided inside RV_find_object_by_path.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Matthew Larson
+ *              April, 2023
+ */
+herr_t RV_copy_object_URI_and_domain_callback(char *HTTP_response, void *callback_data_in, void *callback_data_out) {
```

Comment:
It will document the intent of the callback much better to use a struct type containing the two pointers needed here and then pass a pointer to one of those structs in callback_data_out and cast `callback_data_out` to a pointer to that struct type here.